### DISCUSSION TITLE
Contain drawer within a container

### DISCUSSION THREAD

**Discussion by SumitNalavade** [saadeghi/daisyui#2067] [Q&A]
I'm trying to use the drawer component to create a sidebar menu and I want to contain it within a container like such:

![Screenshot 2023-06-27 at 1 30 12 AM](https://github.com/saadeghi/daisyui/assets/48499839/3e6ca578-3b69-4389-b166-183703413790)

However, when using the component, I'm getting a drawer like this:

![Screenshot 2023-06-27 at 1 31 59 AM](https://github.com/saadeghi/daisyui/assets/48499839/6d2b8a97-1210-4621-a99d-439f1890d129)

Is there any way i can contain the drawer within the smaller gray container?

Thanks!

**Comment by saadeghi** [comment]
Can you share your code on a [Tailwind Play](https://daisyui.com/tailwindplay/) page or StackBlitz or a minimal GitHub repo so I can see and run your code and see what's the issue?

**Reply by SumitNalavade** [reply]
@saadeghi 
Here's the link to a minimal Github Repo to demonstrate:

https://github.com/SumitNalavade/ichild-stackblitz

In the SidePanel.tsx file I've commented out the current side panel implementation I have which looks like this:
<img src="https://github.com/saadeghi/daisyui/assets/48499839/c662329d-b2ba-408d-92e9-2ba158ced7cd" width="500" />

I added the drawer component from DaisyUI along with a button to toggle the drawer as such:
<img src="https://github.com/saadeghi/daisyui/assets/48499839/7380784a-b4cf-4fb7-ade8-4efef0d02820" width="500" />

When the button is clicked, the drawer comes in from the side taking up the full height of the page as such:
<img src="https://github.com/saadeghi/daisyui/assets/48499839/059a2028-5810-4740-92b0-53c834c2171c)" width="500" />

I'm trying to fit this drawer inside the inner container like shown in the first image rather than taking up the full height.

Thanks for the help!

**Comment by jtlindsey** [comment]
@saadeghi I'm seeing the same issue.  Here is another minimum production example you can paste in tailwind play and see by default the drawer overlays the whole page.

```vue
<!-- steps -->
<ul class="steps my-4 w-full">
  <li class="step step-primary">Register</li> 
  <li class="step step-primary">Choose plan</li> 
  <li class="step">Purchase</li> 
  <li class="step">Receive Product</li>
</ul>

<!-- drawer i want to stay inside this div bubble -->
<div class="drawer-should-stay-contained-inside-this-div">
  <div class="drawer">
    <input id="my-drawer" type="checkbox" class="drawer-toggle" />
    <div class="drawer-content">
      <!-- Page content here -->
      <label for="my-drawer" class="btn btn-primary drawer-button">Open drawer</label>
    </div> 
    <div class="drawer-side">
      <label for="my-drawer" class="drawer-overlay"></label>
      <ul class="menu p-4 w-80 h-full bg-base-200 text-base-content">
        <!-- Sidebar content here -->
        <li><a>Sidebar Item 1</a></li>
        <li><a>Sidebar Item 2</a></li>
        
      </ul>
    </div>
  </div>
</div>

```

**Comment by jtlindsey** [comment]
A quick/dirty solution is using `style="position: absolute !important;"` on `drawer-side` like: `<div class="drawer-side" style="position: absolute !important;">`

Better yet `<div class="drawer-side absolute">` works fine.

**Reply by SumitNalavade** [reply]
Works for me with a couple tweaks. Thanks!
